File: frontend/src/api/kakao.ts
import { apiFetch } from "./client";

export interface StudentOption {
  student_id: number;
  name: string;
}

export interface DraftItem {
  queue_id: number;
  student_id: number;
  student_name: string;
  target_phone: string;
  ai_draft: string | null;
  instructor_note: string | null;
  scheduled_at: string;
}

export interface ApprovedItem {
  queue_id: number;
  scheduled_at: string;
  student_name: string;
  target_phone: string;
  instructor_note: string | null;
}

export interface SendLogItem {
  queue_id: number;
  status: string;
  student_name: string;
  target_phone: string;
  sent_at: string | null;
  solapi_msg_id: string | null;
  error_log: string | null;
}

export interface MessageResponse {
  ok: boolean;
  message: string | null;
}

export function fetchKakaoStudents(): Promise<StudentOption[]> {
  return apiFetch("/api/kakao/students");
}

export function createDraft(req: {
  student_id: number;
  target_phone: string;
  template_code: string;
  scheduled_at: string;
}): Promise<MessageResponse> {
  return apiFetch("/api/kakao/drafts", { method: "POST", body: JSON.stringify(req) });
}

export function fetchDrafts(): Promise<DraftItem[]> {
  return apiFetch("/api/kakao/drafts");
}

export function approveDraft(queueId: number, instructorNote: string): Promise<MessageResponse> {
  return apiFetch(`/api/kakao/drafts/${queueId}/approve`, {
    method: "POST",
    body: JSON.stringify({ instructor_note: instructorNote }),
  });
}

export function deleteDraft(queueId: number): Promise<MessageResponse> {
  return apiFetch(`/api/kakao/drafts/${queueId}`, { method: "DELETE" });
}

export function fetchApproved(): Promise<ApprovedItem[]> {
  return apiFetch("/api/kakao/approved");
}

export function revertDraft(queueId: number): Promise<MessageResponse> {
  return apiFetch(`/api/kakao/approved/${queueId}/revert`, { method: "POST" });
}

export function fetchSendLogs(): Promise<SendLogItem[]> {
  return apiFetch("/api/kakao/send-logs");
}

File: backend/app/schemas/kakao.py
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel


class StudentOption(BaseModel):
    student_id: int
    name: str


class CreateDraftRequest(BaseModel):
    student_id: int
    target_phone: str
    template_code: str = "WEEKLY_REPORT_V1"
    scheduled_at: str


class DraftItem(BaseModel):
    queue_id: int
    student_id: int
    student_name: str
    target_phone: str
    ai_draft: Optional[str] = None
    instructor_note: Optional[str] = None
    scheduled_at: str


class ApproveDraftRequest(BaseModel):
    instructor_note: str


class ApprovedItem(BaseModel):
    queue_id: int
    scheduled_at: str
    student_name: str
    target_phone: str
    instructor_note: Optional[str] = None


class SendLogItem(BaseModel):
    queue_id: int
    status: str
    student_name: str
    target_phone: str
    sent_at: Optional[str] = None
    solapi_msg_id: Optional[str] = None
    error_log: Optional[str] = None


class MessageResponse(BaseModel):
    ok: bool
    message: Optional[str] = None
